### PR TITLE
add editable footer snippet

### PR DESCRIPTION
**Pull request by istride**
Related to issue #19 . A fully editable footer snippet that allows changes to navigation links, logos and the "essential" links to privacy policy, terms of use, etc. All links are to pages within the site.

### CODE REVIEW

File: home/models.py [hunk 1]
```diff
@@ -26,3 +27,31 @@ class Article(Page):
     content_panels = Page.content_panels + [
         StreamFieldPanel('body')
     ]
+
+@register_snippet
+class Footer(models.Model):
+    title = models.CharField(max_length=255, blank=True)
+    logos = StreamField([
+        ('image', ImageChooserBlock())
+    ], null=True)
```
Comment (isedwards) [inline home/models.py:36]: The `logos`, `navigation` and `essential` fields are currently all mandatory required fields (despite your `null=True`)...

The confusion is discussed [here](https://github.com/wagtail/wagtail/issues/4306).

In addition to `null=True`, `blank=True` and `required=True`... both `StreamField` and `StreamBlock` have `min_num` and `max_num` arguments (discussed [here](https://github.com/wagtail/wagtail/issues/5175)).

**Review by isedwards — CHANGES_REQUESTED** [review]
Great start of getting footers added.  I've added a couple of initial comments.

File: home/templatetags/home_tags.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from django import template
+from home.models import Footer
+
+register = template.Library()
+
+@register.inclusion_tag('home/tags/footer.html', takes_context=True)
+def footer(context):
+    return {
+        'footer': Footer.objects.all()[0],
```
[the remark this answers is not in the document]
Comment (istride) [reply home/templatetags/home_tags.py]: Yes, worth tidying this up. I expect we'll come up with a better mechanism for choosing the correct footer later on.

File: home/models.py (hunk 1 above)
Comment (istride) [reply home/models.py]: quite confusing. i've attempted to make only `title` and `essential` required fields, the rest are optional.

**Pull request merged**